From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Artur Szafraniec <ganggg111@gmail.com>
Date: Wed, 19 Aug 2026 01:57:04 +0200
Subject: [PATCH] dt-bindings: display: panel: Add Samsung S6E3HA9

Add a binding for the Samsung S6E3HA9 DDIC driving the AMB644WQ01
1440x3040 command mode AMOLED panel, found in the Galaxy S10 family. The
supply names follow the sibling S6E3HA8 binding; on beyond2lte the rails
come from the S2DOS05 display PMIC.

Signed-off-by: Artur Szafraniec <ganggg111@gmail.com>
---
 .../display/panel/samsung,s6e3ha9.yaml        | 71 +++++++++++++++++++
 1 file changed, 71 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/samsung,s6e3ha9.yaml

diff --git a/Documentation/devicetree/bindings/display/panel/samsung,s6e3ha9.yaml b/Documentation/devicetree/bindings/display/panel/samsung,s6e3ha9.yaml
new file mode 100644
index 00000000000..303e57ff622
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/samsung,s6e3ha9.yaml
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/samsung,s6e3ha9.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Samsung S6E3HA9 AMOLED DSI panel
+
+description:
+  The S6E3HA9 is a 1440x3040 DSC-compressed command mode AMOLED panel from
+  Samsung Display (module AMB644WQ01), found in the Galaxy S10 family.
+
+maintainers:
+  - Artur Szafraniec <ganggg111@gmail.com>
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    const: samsung,s6e3ha9
+
+  reg:
+    maxItems: 1
+
+  vdd3-supply:
+    description: VDD3 regulator, 1.8 V
+
+  vci-supply:
+    description: VCI regulator, 3.0 V
+
+  vddr-supply:
+    description: VDDR regulator, 1.5-1.6 V
+
+required:
+  - compatible
+  - reg
+  - port
+  - reset-gpios
+  - vdd3-supply
+  - vci-supply
+  - vddr-supply
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    dsi {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        panel@0 {
+            compatible = "samsung,s6e3ha9";
+            reg = <0>;
+
+            vdd3-supply = <&s2dos05_ldo1>;
+            vci-supply = <&s2dos05_ldo4>;
+            vddr-supply = <&s2dos05_buck1>;
+
+            reset-gpios = <&gpa3 4 GPIO_ACTIVE_LOW>;
+
+            port {
+                panel_in: endpoint {
+                    remote-endpoint = <&dsim_out>;
+                };
+            };
+        };
+    };
+...
